book/src/opaque_type_safety.md

```diff
@@ -0,0 +1,155 @@
+# Opaque type safety
+
+## Ownership and GC
+
+From the moment an opaque type is passed to Dart, it has full ownership of it.
+Dart implements a finalizer for opaque types, but
+the memory of opaque types is not monitoring by Dart and can accumulate, so
+in an instance of an opaque type, 'dispose' must occur.
+
+
+## Opaque type like funtion args
+
+When calling a function with an opaque type argument, Dart threadsafely shares ownership of the opaque type with Rust. This is safe because (Opaque<T>) type T requires Send, Sync, also Rust Opaque<T> only provides Deref (an immutable reference to T).
+If dispose is called on the Dart side before the function call completes, Rust takes full ownership.
+
+
+## Edxample
+ 
+### Case 1 simple call.
+```
+rust: pub fn rust_example() -> Opaque<T>
+rust: pub fn rust_call_example(data: Opaque<T>)
+
+// Arc 1 Dart has full ownership.
+dart: var opaque = var data = await api.rustExample();
+
+// Arc 2 for the duration of the function and after Arc 1. 
+// Dart and Rust share the opaque type.
+dart: await api.rustCallExample(opaque);
+
+// Arc 0 drop opaque type
```

Comment:
What is "Arc 0/1/2"? Are they index, or arc current counter?

Reply:
Arc current counter. Fixed